e2e/nomostest/registryproviders/artifact_registry.go

```diff
@@ -199,12 +210,26 @@ func (a *ArtifactRegistryOCIProvider) DeleteImage(imageName, digest string) erro
 	return a.deleteImage(imageName, digest)
 }
 
+// Reset the state of the registry by deleting images pushed during the test.
+func (a *ArtifactRegistryOCIProvider) Reset() error {
```

Comment:
JFYI the reason the cleanup was factored the way it was before was to avoid sprawl of reimplementing cleanup for each registry variant. But I'd prefer to take a surgical approach and get things working before another refactor